### PR TITLE
Bug 1158388 - [NFC] Swiping up causes jerky, stuttered animation in the Shrinking UI.

### PR DESCRIPTION
**Pull request by cctuan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1158388

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3b7f5bfdaeaa6d7671f9613b2fffb37fd43e26fa

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d849a5343b7764ca0a89b88af4914ceea3dacdb4

File: shared/js/shrinking_ui.js [hunk 1]
```diff
@@ -313,21 +312,27 @@
...
+        this.elements.foregroundElement.style.transitionDuration = '0s';
+        if ('undefined' !== typeof callback) {
+          callback();
+        }
+      }).bind(this);
       if ('TOP' === y) {
+        foregroundElement.style.transitionDuration = '0.3s';
         y = foregroundElement.parentElement.clientHeight;
       } else if ('BOTTOM' === y) {
+        if (this.state.touch.prevY > 0) {
```
Comment (snowmantw) [inline shared/js/shrinking_ui.js]: We should add comments to explain why the transitionDuration jumps between 0.0s and 0.3s, and what the `this.state.touch.preY > 0` means.

File: shared/js/shrinking_ui.js [hunk 2]
```diff
@@ -313,21 +312,27 @@
   ShrinkingUI.prototype._sendingSlideTo =
     function su_sendingSlideTo(y, callback) {
       var foregroundElement = this.elements.foregroundElement;
+      var cbDone = (function on_cbDone(evt) {
+        foregroundElement.removeEventListener('transitionend', cbDone);
+        this.elements.foregroundElement.style.transitionDuration = '0s';
```
Comment (snowmantw) [inline shared/js/shrinking_ui.js]: Comment to explain why it should be 0s here.

File: shared/js/shrinking_ui.js [hunk 3]
```diff
@@ -313,21 +312,27 @@
...
+        if ('undefined' !== typeof callback) {
+          callback();
+        }
+      }).bind(this);
       if ('TOP' === y) {
+        foregroundElement.style.transitionDuration = '0.3s';
```
Comment (snowmantw) [inline shared/js/shrinking_ui.js]: Comment to explain why it should be 0.3s here.

File: shared/js/shrinking_ui.js [hunk 4]
```diff
@@ -493,6 +498,7 @@
    */
   ShrinkingUI.prototype._handleSendingStart =
     function su_handleSendingStart(evt) {
+      this.state.touch.initY = evt.touches[0].pageY;
```
Comment (snowmantw) [inline shared/js/shrinking_ui.js]: Comment to explain why we need this: for keeping the position the finger start to touch on the sliding cover, and we would use it to calculate the transition offset every frame.

File: shared/js/shrinking_ui.js [hunk 5]
```diff
@@ -517,7 +523,7 @@
   ShrinkingUI.prototype._handleSendingSlide =
     function su_handleSendingSlide(evt) {
       var pgy = evt.touches[0].pageY;
-      var slideY = this.elements.backgroundElement.clientHeight - pgy;
+      var slideY = this.state.touch.initY  - pgy;
```
Comment (snowmantw) [inline shared/js/shrinking_ui.js]: You can comment it at above or here.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3b794da8cdd855183e39a432968d52a65295f7bd

File: shared/js/shrinking_ui.js (hunk 5 above)
Comment (cctuan) [inline shared/js/shrinking_ui.js]: commented above.

**Pull request merged**